cookbook/.cursor/rules/cookbook-recipe-subscriptions.mdc

```diff
@@ -157,6 +158,310 @@ export function SellingPlanSelector({
+  query SubscriptionsContractsQuery {
+    customer {
+      subscriptionContracts(first: 100) {
+        nodes {
+          ...SubscriptionContract
+          lines(first: 100) {
+            nodes {
+              name
+              id
+            }
+          }
+        }
+      }
+    }
+  }
+  ${SUBSCRIPTION_CONTRACT_FRAGMENT}
+` as const;
+
+```
+
+### templates/skeleton/app/routes/account.subscriptions.tsx
+
+Subscriptions management page.
+
+```tsx
+import type {
+  CustomerFragment,
+  SubscriptionBillingPolicyFragment,
+} from 'customer-accountapi.generated';
+import {
+  data,
+  LinksFunction,
+  type ActionFunctionArgs,
+  type LoaderFunctionArgs,
+} from '@shopify/remix-oxygen';
+import {
+  Form,
+  useActionData,
+  useLoaderData,
+  useNavigation,
+  type MetaFunction,
+} from '@remix-run/react';
+import {
+  SUBSCRIPTION_CANCEL_MUTATION,
+  SUBSCRIPTIONS_CONTRACTS_QUERY,
+} from '../graphql/customer-account/CustomerDetailsQuery';
+
+import accountSubscriptionsStyle from '~/styles/account-subscriptions.css?url';
+
+export type ActionResponse = {
+  error: string | null;
+};
+
+export const meta: MetaFunction = () => {
+  return [{title: 'Subscriptions'}];
+};
+
+export const links: LinksFunction = () => [
+  {rel: 'stylesheet', href: accountSubscriptionsStyle},
+];
+
+export async function loader({context}: LoaderFunctionArgs) {
+  await context.customerAccount.handleAuthStatus();
+
+  const {data: subscriptions} = await context.customerAccount.query(
+    SUBSCRIPTIONS_CONTRACTS_QUERY,
+  );
+
+  return {subscriptions};
+}
+
+export async function action({request, context}: ActionFunctionArgs) {
+  const {customerAccount} = context;
+
+  if (request.method !== 'DELETE') {
+    return data({error: 'Method not allowed'}, {status: 405});
+  }
+
+  const form = await request.formData();
+
+  try {
+    const subId = form.get('subId');
+
+    if (!subId) {
+      throw new Error('Subscription ID is required');
+    }
+
+    await customerAccount.mutate(SUBSCRIPTION_CANCEL_MUTATION, {
+      variables: {
+        subscriptionContractId: subId.toString(),
+      },
+    });
+
+    return {
+      error: null,
+    };
+  } catch (error: any) {
+    return data(
+      {
+        error: error.message,
+      },
+      {
+        status: 400,
+      },
+    );
+  }
+}
+
+export default function AccountProfile() {
+  const {state} = useNavigation();
+  const action = useActionData<ActionResponse>();
+
+  const {subscriptions} = useLoaderData<typeof loader>();
+
+  const sortedSubscriptions =
+    subscriptions?.customer?.subscriptionContracts.nodes.sort((a, b) => {
+      // active subscriptions first
+      if (a.status === 'ACTIVE' && b.status !== 'ACTIVE') {
+        return -1;
+      }
+      if (a.status !== 'ACTIVE' && b.status === 'ACTIVE') {
+        return 1;
+      }
+      // then, newest first
+      return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+    });
```

Comment:
In my opinion we should keep the example simpler and let people add sorting if they want.